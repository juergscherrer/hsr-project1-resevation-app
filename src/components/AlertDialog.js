import React from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';

class AlertDialog extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      open: this.props.open
    };
  }

  componentDidUpdate(prevProps) {
    if (this.props.open !== prevProps.open) {
      this.setState({ open: this.props.open });
    }
  }

  handleFalse = () => {
    this.setState({ open: false });
    this.props.handleAnswer(false);
  };

  handleTrue = () => {
    this.setState({ open: false });
    this.props.handleAnswer(true);
  };

  render() {
    const { text } = this.props;
    return (
      <div>
        <Dialog
          open={this.state.open}
          onClose={this.handleFalse}
          aria-labelledby="alert-dialog-title"
          aria-describedby="alert-dialog-description"
        >
          <DialogContent>
            <DialogContentText id="alert-dialog-description">
              {text}
            </DialogContentText>
          </DialogContent>
          <DialogActions>
            <Button id="no" onClick={this.handleFalse} color="primary">
              Nein
            </Button>
            <Button
              id="yes"
              onClick={this.handleTrue}
              color="primary"
              autoFocus
            >
              Ja
            </Button>
          </DialogActions>
        </Dialog>
      </div>
    );
  }
}

export default AlertDialog;
