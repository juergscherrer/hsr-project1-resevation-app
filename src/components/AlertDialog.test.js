import React from 'react';
import { shallow } from 'enzyme';
import AlertDialog from './AlertDialog';

let mockHandleAnswer = answer => {
  if (answer) {
    return true;
  } else {
    return false;
  }
};

describe('<AlertDialog />', () => {
  it('should be defined', () => {
    expect(AlertDialog).toBeDefined();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <AlertDialog
        open="true"
        text="Test Text"
        handleAnswer={mockHandleAnswer}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a text prop', () => {
    const wrapper = shallow(
      <AlertDialog
        open="true"
        text="Test Text"
        handleAnswer={mockHandleAnswer}
      />
    );

    expect(typeof wrapper.instance().props.text).toBe('string');
    expect(wrapper.instance().props.text).toEqual('Test Text');
  });

  it('should have a open prop', () => {
    const wrapper = shallow(
      <AlertDialog
        open="false"
        text="Test Text"
        handleAnswer={mockHandleAnswer}
      />
    );

    expect(typeof wrapper.instance().props.open).toBe('string');
    expect(wrapper.instance().props.open).toEqual('false');
  });

  it('should call handleAnswer and return true', () => {
    const wrapper = shallow(
      <AlertDialog
        open="false"
        text="Test Text"
        handleAnswer={mockHandleAnswer('test answer')}
      />
    );

    expect(typeof wrapper.instance().props.handleAnswer).toBeTruthy();
  });
});
