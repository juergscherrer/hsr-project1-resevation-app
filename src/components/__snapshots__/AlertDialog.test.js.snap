// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertDialog /> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AlertDialog
    handleAnswer={[Function]}
    open="true"
    text="Test Text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <WithStyles(Dialog)
        aria-describedby="alert-dialog-description"
        aria-labelledby="alert-dialog-title"
        onClose={[Function]}
        open="true"
      >
        <WithStyles(DialogContent)>
          <WithStyles(DialogContentText)
            id="alert-dialog-description"
          >
            Test Text
          </WithStyles(DialogContentText)>
        </WithStyles(DialogContent)>
        <WithStyles(DialogActions)>
          <WithStyles(Button)
            color="primary"
            id="no"
            onClick={[Function]}
          >
            Nein
          </WithStyles(Button)>
          <WithStyles(Button)
            autoFocus={true}
            color="primary"
            id="yes"
            onClick={[Function]}
          >
            Ja
          </WithStyles(Button)>
        </WithStyles(DialogActions)>
      </WithStyles(Dialog)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "aria-describedby": "alert-dialog-description",
        "aria-labelledby": "alert-dialog-title",
        "children": Array [
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              id="alert-dialog-description"
            >
              Test Text
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>,
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              id="no"
              onClick={[Function]}
            >
              Nein
            </WithStyles(Button)>
            <WithStyles(Button)
              autoFocus={true}
              color="primary"
              id="yes"
              onClick={[Function]}
            >
              Ja
            </WithStyles(Button)>
          </WithStyles(DialogActions)>,
        ],
        "onClose": [Function],
        "open": "true",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(DialogContentText)
              id="alert-dialog-description"
            >
              Test Text
            </WithStyles(DialogContentText)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Test Text",
              "id": "alert-dialog-description",
            },
            "ref": null,
            "rendered": "Test Text",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <WithStyles(Button)
                color="primary"
                id="no"
                onClick={[Function]}
              >
                Nein
              </WithStyles(Button)>,
              <WithStyles(Button)
                autoFocus={true}
                color="primary"
                id="yes"
                onClick={[Function]}
              >
                Ja
              </WithStyles(Button)>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Nein",
                "color": "primary",
                "id": "no",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Nein",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoFocus": true,
                "children": "Ja",
                "color": "primary",
                "id": "yes",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Ja",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <WithStyles(Dialog)
          aria-describedby="alert-dialog-description"
          aria-labelledby="alert-dialog-title"
          onClose={[Function]}
          open="true"
        >
          <WithStyles(DialogContent)>
            <WithStyles(DialogContentText)
              id="alert-dialog-description"
            >
              Test Text
            </WithStyles(DialogContentText)>
          </WithStyles(DialogContent)>
          <WithStyles(DialogActions)>
            <WithStyles(Button)
              color="primary"
              id="no"
              onClick={[Function]}
            >
              Nein
            </WithStyles(Button)>
            <WithStyles(Button)
              autoFocus={true}
              color="primary"
              id="yes"
              onClick={[Function]}
            >
              Ja
            </WithStyles(Button)>
          </WithStyles(DialogActions)>
        </WithStyles(Dialog)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "aria-describedby": "alert-dialog-description",
          "aria-labelledby": "alert-dialog-title",
          "children": Array [
            <WithStyles(DialogContent)>
              <WithStyles(DialogContentText)
                id="alert-dialog-description"
              >
                Test Text
              </WithStyles(DialogContentText)>
            </WithStyles(DialogContent)>,
            <WithStyles(DialogActions)>
              <WithStyles(Button)
                color="primary"
                id="no"
                onClick={[Function]}
              >
                Nein
              </WithStyles(Button)>
              <WithStyles(Button)
                autoFocus={true}
                color="primary"
                id="yes"
                onClick={[Function]}
              >
                Ja
              </WithStyles(Button)>
            </WithStyles(DialogActions)>,
          ],
          "onClose": [Function],
          "open": "true",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(DialogContentText)
                id="alert-dialog-description"
              >
                Test Text
              </WithStyles(DialogContentText)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Test Text",
                "id": "alert-dialog-description",
              },
              "ref": null,
              "rendered": "Test Text",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(Button)
                  color="primary"
                  id="no"
                  onClick={[Function]}
                >
                  Nein
                </WithStyles(Button)>,
                <WithStyles(Button)
                  autoFocus={true}
                  color="primary"
                  id="yes"
                  onClick={[Function]}
                >
                  Ja
                </WithStyles(Button)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Nein",
                  "color": "primary",
                  "id": "no",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Nein",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoFocus": true,
                  "children": "Ja",
                  "color": "primary",
                  "id": "yes",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Ja",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
